unified-codes/libs/ui/src/lib/components/entity/organisms/EntityBrowser.stories.tsx

```diff
@@ -304,21 +299,60 @@ export const withProps = () => {
     />
   ), [classes, input, label, onChange, onClear, onSearch]);
 
-  // Type filter.
+  // Toggle bar.
 
-  const { types } = useTypeFilterState();
+  const { 
+    filterByDrug,
+    filterByUnitOfUse, 
+    filterByOther,
+    setFilterByDrug, 
+    setFilterByUnitOfUse, 
+    setFilterByOther
+  } = useToggleBarState();
+
+  const onToggle = React.useCallback((type: EEntityType) => {
+    switch(type) {
+      case EEntityType.DRUG: {
+        setFilterByDrug(!filterByDrug);
+        break;
+      }
+      case EEntityType.UNIT_OF_USE: {
+        setFilterByUnitOfUse(!filterByUnitOfUse);
+        break;
+      }
+      case EEntityType.OTHER: {
+        setFilterByOther(!filterByOther);
+        break;
+      }
+    }
+  }, [filterByDrug, filterByUnitOfUse, filterByOther, setFilterByDrug, setFilterByUnitOfUse, setFilterByOther]);
 
-  const entityBrowserTypeFilter = React.useMemo(() => (
-    <EntityTypeFilter 
+  const entityBrowserToggleBar = React.useMemo(() => (
+    <EntityToggleBar 
       classes={{ 
-        root: classes?.typeFilter?.root, 
-        toggleButtonActive: classes?.typeFilter?.toggleButtonActive,
-        toggleButtonInactive: classes?.typeFilter?.toggleButtonInactive,
-        toggleButtonGroup: classes?.typeFilter?.toggleButtonGroup,
+        root: classes?.toggleBar?.root, 
+        buttonContainer: classes?.toggleBar?.buttonContainer,
+        buttonActive: classes?.toggleBar?.buttonActive,
+        buttonInactive: classes?.toggleBar?.buttonInactive,
+      }}
+      buttonTypes={[
+        EEntityType.DRUG,
+        EEntityType.UNIT_OF_USE,
+        EEntityType.OTHER
+      ]}
+      buttonStates={{
+        [EEntityType.DRUG]: filterByDrug,
+        [EEntityType.UNIT_OF_USE]: filterByUnitOfUse,
+        [EEntityType.OTHER]: filterByOther
+      }}
+      buttonLabels={{
+        [EEntityType.DRUG]: 'Drug',
+        [EEntityType.UNIT_OF_USE]: 'Unit of use',
+        [EEntityType.OTHER]: 'Other'
```

Comment:
As above, could lift up the state a bit more and just replace all this with `toggleButtons={toggleButtons}`!